clusterloader2/testing/load/modules/scheduler-throughput.yaml

```diff
@@ -0,0 +1,113 @@
+## Input params
+# Valid actions: "create", "delete"
+{{$action := .action}}
+{{$namespaces := .namespaces}}
+{{$replicasPerNamespace := .replicasPerNamespace}}
+{{$schedulerThroughputNamespaces := .schedulerThroughputNamespaces}}
+{{$schedulerThroughputPodsPerDeployment := .schedulerThroughputPodsPerDeployment}}
+
+## Derivative variables
+{{$is_creating := (eq .action "create")}}
+{{$is_deleting := (eq .action "delete")}}
+
+## CL2 params
+{{$ENABLE_HUGE_SERVICES := DefaultParam .CL2_ENABLE_HUGE_SERVICES false}}
+{{$SCHEDULER_THROUGHPUT_THRESHOLD := DefaultParam .CL2_SCHEDULER_THROUGHPUT_THRESHOLD 100}}
+
+steps:
+{{if $is_creating}}
+- name: Creating scheduler throughput measurements
+  measurements:
+  - Identifier: HighThroughputPodStartupLatency
+    Method: PodStartupLatency
+    Params:
+      action: start
+      labelSelector: group = scheduler-throughput
+      threshold: 1h # TODO(https://github.com/kubernetes/perf-tests/issues/1024): Ideally, this should be 5s
+  - Identifier: WaitForSchedulerThroughputDeployments
+    Method: WaitForControlledPodsRunning
+    Params:
+      action: start
+      apiVersion: apps/v1
+      kind: Deployment
+      labelSelector: group = scheduler-throughput
+      # The operation timeout shouldn't be less than 20m to make sure that ~10m node
+      # failure won't fail the test. See https://github.com/kubernetes/kubernetes/issues/73461#issuecomment-467338711
+      operationTimeout: 20m
+  - Identifier: SchedulingThroughput
+    Method: SchedulingThroughput
+    Params:
+      action: start
+      labelSelector: group = scheduler-throughput
+      measurmentInterval: 1s
+  {{if $ENABLE_HUGE_SERVICES}}
+- name: Creating huge services
+  phases:
+  - namespaceRange:
+      min: {{AddInt $namespaces 1}}
+      max: {{AddInt $namespaces $schedulerThroughputNamespaces}}
+    replicasPerNamespace: 1
+    tuningSet: Sequence
+    objectBundle:
+    - basename: huge-service
+      objectTemplatePath: service.yaml
+  {{end}}
+{{end}}
+- name: {{$action}} scheduler throughput pods
+  phases:
+  - namespaceRange:
+      min: {{AddInt $namespaces 1}}
+      max: {{AddInt $namespaces $schedulerThroughputNamespaces}}
+    replicasPerNamespace: {{$replicasPerNamespace}}
+    tuningSet: SchedulerThroughputParallel
+    objectBundle:
+    - basename: scheduler-throughput-deployment
+      objectTemplatePath: simple-deployment.yaml
+{{if $is_creating}}
```

Comment:
You can avoid this if by simply putting templateFillMap unconditionally. It is simply ignored for deletion (i.e. the TemplateProvider is using some hack to avoid passing templateFillMap here, see https://github.com/kubernetes/perf-tests/blob/master/clusterloader2/pkg/config/template_provider.go#L90-L95

Reply:
Done. Also thanks for referencing the code! :)